### ISSUE TITLE
BOUT++-v4.3.1 some examples can not run

### ISSUE THREAD

**Issue by wphu** [boutproject/BOUT-dev#2041]
Hello,

In BOUT++-v4.3.1 examples, uedge-benchmark and tokamak-2fluid can be compiled successfully, but Error encountered when running, errors are shown below:

Error encountered
====== Exception path ======
[bt] #8 ./ue_bmark() [0x439e3a]
_start 于 ??:?
[bt] #7 /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xe7) [0x7fc9e07a1a87]
__libc_start_main 于 /build/glibc-kAz5Pl/glibc-2.27/csu/../csu/libc-start.c:344
[bt] #6 ./ue_bmark() [0x433c35]
main 于 /home/huwanpeng/opt-gcc/gcc/include/c++/8.1.0/bits/unique_ptr.h:81
(已内连入)?? 于 /home/huwanpeng/opt-gcc/gcc/include/c++/8.1.0/bits/unique_ptr.h:274
(已内连入)main 于 /home/huwanpeng/codes/BOUT++-v4.3.1/examples/uedge-benchmark/ue_bmark.cxx:253
[bt] #5 ./ue_bmark() [0x474510]
Solver::solve(int, double) 于 /home/huwanpeng/codes/BOUT++-v4.3.1/src/solver/solver.cxx:476
[bt] #4 ./ue_bmark() [0x472905]
Solver::run_rhs(double) 于 /home/huwanpeng/codes/BOUT++-v4.3.1/src/solver/solver.cxx:1161
[bt] #3 ./ue_bmark() [0x44040f]
UedgeBenchmark::rhs(double) 于 /home/huwanpeng/codes/BOUT++-v4.3.1/examples/uedge-benchmark/ue_bmark.cxx:240 (discriminator 5)
[bt] #2 ./ue_bmark() [0x4bea74]
Div_par_K_Grad_par(Field3D const&, Field3D const&, CELL_LOC) 于 /home/huwanpeng/codes/BOUT++-v4.3.1/src/mesh/difops.cxx:308 (discriminator 18)
[bt] #1 ./ue_bmark() [0x4c1933]
Field3D const interp_to<Field3D>(Field3D const&, CELL_LOC, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) 于 /home/huwanpeng/codes/BOUT++-v4.3.1/src/mesh/../../include/interpolation.hxx:66 (discriminator 3)
====== Back trace ======
 -> Ncxx4::close on line 266 of 'ncxx4.cxx'
 -> const T interp_to(const T&, CELL_LOC, std::__cxx11::string) [with T = Field3D; std::__cxx11::string = std::__cxx11::basic_stri

====== Exception thrown ======
Assertion failed in ../../include/interpolation.hxx, line 66: loc != CELL_DEFAULT

Thanks, looking forward to reply.

**Comment by bshanahan** [comment]
Hi, this looks to be a known bug which was fixed yesterday.  See #[2038](https://github.com/boutproject/BOUT-dev/pull/2038).  Try `git pull` on the master branch and recompiling.

**Comment by bshanahan** [comment]
Sorry, this has only been fixed in the `next` branch.  It should be going into `master` shortly -- I will reopen until it gets merged.

**Comment by ZedThree** [comment]
Now fixed in `master` in #2039. @wphu please could you confirm that the latest commit on `master` solves this?

**Comment by wphu** [comment]
@ZedThree @bshanahan  Thanks for quick reply. I git clone the master branch, rand compile. uedge-benchmark is OK, but tokamak-2fluid still can not run.

**Comment by bshanahan** [comment]
Can you post the error for tokamak-2fluid?

**Comment by wphu** [comment]
the error:

```
Error encountered
====== Exception path ======
[bt] #6 ./2fluid() [0x43a70a]
_start 于 ??:?
[bt] #5 /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xe7) [0x7fd711fe8a87]
__libc_start_main 于 /build/glibc-kAz5Pl/glibc-2.27/csu/../csu/libc-start.c:344
[bt] #4 ./2fluid() [0x4344ea]
main 于 /home/huwanpeng/opt-gcc/gcc/include/c++/8.1.0/bits/unique_ptr.h:831
(已内连入)main 于 /home/huwanpeng/codes/BOUT-dev/examples/tokamak-2fluid/2fluid.cxx:896
[bt] #3 ./2fluid() [0x49c428]
Solver::setModel(PhysicsModel*) 于 /home/huwanpeng/codes/BOUT-dev/src/solver/../../include/bout/physicsmodel.hxx:79
(已内连入)Solver::setModel(PhysicsModel*) 于 /home/huwanpeng/codes/BOUT-dev/src/solver/solver.cxx:73
[bt] #2 ./2fluid() [0x449a02]
TwoFluid::init(bool) 于 /home/huwanpeng/codes/BOUT-dev/examples/tokamak-2fluid/../../include/field3d.hxx:231
(已内连入)TwoFluid::init(bool) 于 /home/huwanpeng/codes/BOUT-dev/examples/tokamak-2fluid/2fluid.cxx:341
[bt] #1 ./2fluid() [0x462638]
Field::setLocation(CELL_LOC) 于 /home/huwanpeng/codes/BOUT-dev/src/field/field.cxx:66
====== Back trace ======
 -> void Field::setLocation(CELL_LOC) on line 50 of 'field.cxx'
 -> int init(bool)  on line 104 of '2fluid.cxx'

====== Exception thrown ======
Field: Trying to set off-centre location on non-staggered grid
         Did you mean to enable staggered grids?
```

**Comment by johnomotani** [comment]
@wphu the `tokamak-2fluid` example seems to be old, and I have problems getting it to run too (see #2043). Is it important for you to fix this example?

**Comment by wphu** [comment]
@johnomotani the tokamak-2fluid example is not important for me, I am a new user, and want to learn  to simulate sol and divertor. I will focus on uedge-benchmark and hermes-2/3.  So take your time, or just leave it.

**Comment by ZedThree** [comment]
Fixed in #2043
